dot/parachain/candidate-validation/candidate_validation.go

```diff
@@ -86,35 +90,30 @@ func (cv *CandidateValidation) processMessages(wg *sync.WaitGroup) {
 			logger.Debugf("received message %v", msg)
 			switch msg := msg.(type) {
 			case ValidateFromChainState:
-				runtimeInstance, err := cv.BlockState.GetRuntime(msg.CandidateReceipt.Descriptor.RelayParent)
-				if err != nil {
-					logger.Errorf("failed to get runtime: %w", err)
-					msg.Ch <- parachaintypes.OverseerFuncRes[ValidationResult]{
-						Err: err,
-					}
-				}
-				result, err := validateFromChainState(runtimeInstance, msg.Pov, msg.CandidateReceipt)
-				if err != nil {
-					logger.Errorf("failed to validate from chain state: %w", err)
-					msg.Ch <- parachaintypes.OverseerFuncRes[ValidationResult]{
-						Err: err,
-					}
-				} else {
-					msg.Ch <- parachaintypes.OverseerFuncRes[ValidationResult]{
-						Data: *result,
-					}
-				}
+				cv.validateFromChainState(msg)
+
 			case ValidateFromExhaustive:
-				result, err := validateFromExhaustive(cv.ValidationHost, msg.PersistedValidationData,
-					msg.ValidationCode, msg.CandidateReceipt, msg.PoV)
-				if err != nil {
-					logger.Errorf("failed to validate from exhaustive: %w", err)
-					msg.Ch <- parachaintypes.OverseerFuncRes[ValidationResult]{
-						Err: err,
+				taskResult := make(chan *pvf.ValidationTaskResult)
+				validationTask := &pvf.ValidationTask{
+					PersistedValidationData: msg.PersistedValidationData,
+					ValidationCode:          &msg.ValidationCode,
+					CandidateReceipt:        &msg.CandidateReceipt,
+					PoV:                     msg.PoV,
+					ExecutorParams:          msg.ExecutorParams,
+					PvfExecTimeoutKind:      msg.PvfExecTimeoutKind,
+					ResultCh:                taskResult,
+				}
+				go cv.pvfHost.Validate(validationTask)
+
+				result := <-taskResult
```

Comment:
I didn't understand this part, is this meant to be concurrent? From my understanding this blocks the process until it receives the result, so it becomes a sync task and can be done w/o dispatching a goroutine.

```go
result := cv.pvfHost.Validate(validationTask)
```